### ISSUE TITLE
Loss Function

### ISSUE THREAD

**Issue by Rishikesh-Reddy** [Rishikesh-Reddy/Visual-Permutation-Learning#1]
Figure out a way to implement the Loss function.

currently, the loss function returns `NaN` after a few training batches.
